### PR TITLE
create utilities for writing unit tests with a Mock Provider (no rpc eth node required)

### PR DESCRIPTION
**Pull request by insipx**

### CODE REVIEW

File: xps-gateway/src/lib.rs [hunk 1]
```diff
@@ -22,3 +22,29 @@ pub async fn run(host: String, port: u16) -> Result<()> {
...
+    use std::str::FromStr;
+
+    use ethers::{
+        providers::{MockProvider, Provider},
+        types::{Address, U64},
+    };
+
+    use crate::types::GatewayContext;
+
+    pub async fn create_mock_context() -> (GatewayContext<Provider<MockProvider>>, MockProvider) {
```
Comment (jac18281828) [inline xps-gateway/src/lib.rs:37]: This is really cool.  Thanks for leading the way on this!

**Review by jac18281828 — COMMENTED** [review]
What do you think of continuing to provide 'smoke' test coverage from the anvil integration test, while implementing the blanket api coverage in both mock and unit form?

**Comment by insipx** [comment]
> What do you think of continuing to provide 'smoke' test coverage from the anvil integration test, while implementing the blanket api coverage in both mock and unit form?

yup! I think think the coverage we get from integration tests is super valuable, but having the ability to test smaller units of code w/o worrying about the network is v. nice for quick testing

I'm still experimenting with this a bit, because it requires pushing a bunch of requirements that the `ethers` library expects (i.e why this will require some boilerplate for generic transactions to the registry), but it will result in an easier time writing tests for our rpcs.

**Comment by insipx** [comment]
Going to put this on hold while I work on #20. Right now, you can push responses to the mock provider, but I don't think it's very useful, because all the responses have to be carefully curated to match the contract you're calling. In the interest of time and not rewriting the results of contract calls, I want to finish #20 and maybe revist later

it would be nice if we could find a library that can call a solidity contract w/ a mock state, without a node -- not sure if this exists

**Pull request closed without merging**